src/patients/visits/AddVisitModal.tsx

```diff
@@ -1,30 +1,29 @@
 import { Modal } from '@hospitalrun/components'
 import { addMonths } from 'date-fns'
 import React, { useState, useEffect } from 'react'
-import { useDispatch, useSelector } from 'react-redux'
 
 import useTranslator from '../../shared/hooks/useTranslator'
 import Visit from '../../shared/model/Visit'
-import { RootState } from '../../shared/store'
-import { addVisit } from '../patient-slice'
+import useAddVisit from '../hooks/useAddVisit'
 import VisitForm from './VisitForm'
 
 interface Props {
   show: boolean
   onCloseButtonClick: () => void
+  patientId: string
 }
 
 const initialVisitState = {
   startDateTime: new Date().toISOString(),
   endDateTime: addMonths(new Date(), 1).toISOString(),
-}
+} as Visit
```

Comment:
Rather than having to cast this object, we should create a `VisitRequest` object using TypeScript's `Omit`. We would emit any "generated" field such as `id` or `createdOn`